protocol/statetracker/emergency_tracker.go

```diff
@@ -0,0 +1,84 @@
+package statetracker
+
+import (
+	"sync"
+	"time"
+
+	"github.com/lavanet/lava/utils"
+	downtimev1 "github.com/lavanet/lava/x/downtime/v1"
+)
+
+type EmergencyTracker struct {
+	lock             sync.RWMutex
+	downtimeParams   downtimev1.Params
+	virtualEpochsMap map[uint64]uint64
+	latestEpoch      uint64
+	latestEpochTime  time.Time
+}
+
+func (cs *EmergencyTracker) GetVirtualEpoch(epoch uint64) uint64 {
+	cs.lock.RLock()
+	defer cs.lock.RUnlock()
+	return cs.virtualEpochsMap[epoch]
+}
+
+func (cs *EmergencyTracker) SetDowntimeParams(params downtimev1.Params) {
+	cs.lock.Lock()
+	defer cs.lock.Unlock()
+	cs.downtimeParams = params
+}
+
+func (cs *EmergencyTracker) GetDowntimeParams() downtimev1.Params {
+	cs.lock.RLock()
+	defer cs.lock.RUnlock()
+	return cs.downtimeParams
+}
+
+func (cs *EmergencyTracker) UpdateEpoch(epoch uint64) {
```

Comment:
if this is called by the consumer, this will result in a problem, i assume we create two classes inheriting form one another, one for the consumer tracking only the latest the other for the provider, this way we can't confuse it